### PR TITLE
Add xkeyboard-config to depends builds

### PR DESCRIPTION
**Pull request by garbear**
As discussed with @lrusak at devcon, I'm opening a PR for the work I did adding xkeyboard-config to depends a few months ago. This is certainly required on embedded platforms, but most platforms package it themselves. lrusak says people still come to him asking about broken input though, so it would be nice if these people didn't upgrade to broken input. Myself being one of these people, while upgrading Kodi for the Steam Link, hence this patch.

CC @popcornmix, as I think lrusak mentioned you had this issue.

## How Has This Been Tested?
Tested on Steam Link. Confirmed that the files are included, but keyboard input is still untested at the moment.

## Types of change

- [x] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [ ] **Improvement** (non-breaking change which improves existing functionality)
- [x] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

### CODE REVIEW

**Review by lrusak — APPROVED** [review]
I'm not too happy about adding a run time dependency to the depends (as most Linux systems won't use the depends build anyways, at least not for running [read jenkins]) however, to help people to understand why keyboard input may not work under these circumstances I'm happy to have this added in.

**Comment by garbear** [comment]
OK, I'll do some more testing and see if this is complete

File: cmake/modules/FindXkeyboard-config.cmake [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+endif()
+
+include (FindPackageHandleStandardArgs)
+find_package_handle_standard_args(Xkeyboard-config
+  REQUIRED_VARS XKEYBOARD-CONFIG_PREFIX
+  VERSION_VAR XKEYBOARD-CONFIG_VERSION)
+
+find_file(XKEYBOARD-CONFIG_DIR xkb PATHS ${XKEYBOARD-CONFIG_PREFIX}/share PATH_SUFFIXES X11)
+
+if(XKEYBOARD-CONFIG_FOUND)
```
Comment (Rechi) [inline cmake/modules/FindXkeyboard-config.cmake]: XKEYBOARD-CONFIG_DIR could be not found

File: cmake/modules/FindXkeyboard-config.cmake [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+  install(DIRECTORY ${XKEYBOARD-CONFIG_DIR}/ DESTINATION ${CMAKE_INSTALL_FULL_DATAROOTDIR}/X11/xkb)
```
Comment (Rechi) [inline cmake/modules/FindXkeyboard-config.cmake]: I don't think Kodi should install anything to that directory.

File: tools/depends/native/intltool-native/Makefile [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+PREFIX=$(NATIVEPREFIX)
+PLATFORM=$(NATIVEPLATFORM)
+DEPS= ../../Makefile.include Makefile
+
+# lib name, version
+LIBNAME=intltool
+VERSION=0.51.0
+SOURCE=$(LIBNAME)-$(VERSION)
+ARCHIVE=$(SOURCE).tar.gz
+BASE_URL=http://launchpad.net/intltool/trunk/0.51.0/+download/
```
Comment (Rechi) [inline tools/depends/native/intltool-native/Makefile]: upload

File: tools/depends/native/util-macros-native/Makefile [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+PREFIX=$(NATIVEPREFIX)
+PLATFORM=$(NATIVEPLATFORM)
+DEPS= ../../Makefile.include Makefile
+
+# lib name, version
+LIBNAME=util-macros
+VERSION=1.19.2
+SOURCE=$(LIBNAME)-$(VERSION)
+ARCHIVE=$(SOURCE).tar.bz2
+BASE_URL=http://xorg.freedesktop.org/archive/individual/util
```
Comment (Rechi) [inline tools/depends/native/util-macros-native/Makefile]: upload

File: tools/depends/target/xkeyboard-config/Makefile [hunk 5]
```diff
@@ -0,0 +1,34 @@
+include ../../Makefile.include
+DEPS=../../Makefile.include Makefile
+
+# lib name, version
+LIBNAME=xkeyboard-config
+VERSION=2.23.1
+SOURCE=$(LIBNAME)-$(VERSION)
+ARCHIVE=$(SOURCE).tar.bz2
+BASE_URL=http://www.x.org/releases/individual/data/$(LIBNAME)
```
Comment (Rechi) [inline tools/depends/target/xkeyboard-config/Makefile]: upload

**Review by duglaslopez010192 — APPROVED** [review]

**Comment by phunkyfish** [comment]
@garbear do you want to get this merged?

**Comment by garbear** [comment]
Most embedded projects use OSes that provide xkeyboard-config, and the only one I've come across that didn't I'm no longer working on, so lets close this.

**Pull request closed without merging**